### PR TITLE
ignore_transformations when `heic:preserve-orientation=true`

### PR DESCRIPTION
**Pull request by liuzhe0223**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/ImageMagick/ImageMagick/pulls) open
- [x] I have verified that I am following the existing coding patterns and practices as demonstrated in the repository.

### Description

 Previous fix for #1232 is not good enough with photos that has no `heif irot` but only `exif:Orientation`.

The `heic` coder will transform the origin pixels with properties (irot, imir, clap) of heif default.

Example photo in  #1232 has both `heif irot` and `exif:Orientation`.  And the two will rotate origin  pixels with the same degrees. So previous fix got correct result.

This pull-request use `ignore_transformations` decode options  to keep the origin pixels orientation instead of  `CompensateOrientation`.

### Example photo
 https://www.dropbox.com/s/bzkbemy7ngiaghb/FvwrpJFxWoehjaYqtnaXa29TauF8.heic?dl=0
This photo has no `heif irot` but only `exif:Orientation`.

### CODE REVIEW

**Comment by liuzhe0223** [comment]
Thank you for your review , I have fixed code according to your comments @dlemstra .

File: coders/heic.c [hunk 1]
```diff
@@ -331,8 +285,21 @@ static Image *ReadHEICImage(const ImageInfo *image_info,
...
   (void) SetImageColorspace(image,YCbCrColorspace,exception);
+  decode_options=(struct heif_decoding_options *) NULL;
+  option=GetImageOption(image_info,"heic:preserve-orientation");
+  if (IsStringTrue(option) == MagickTrue)
+    {
+      decode_options=heif_decoding_options_alloc();
+      decode_options->ignore_transformations=1;
+    }
+  else
+    {
```
Comment (dlemstra) [inline coders/heic.c]: Almost there 👍 Thanks for making all these changes for us. 

We don't use braces when the statement is just one line so could you remove them here?

Comment (liuzhe0223) [reply coders/heic.c]: ok

Comment (liuzhe0223) [reply coders/heic.c]: done

**Pull request merged**